src/components/MoneyRequestAmountInput.tsx

```diff
@@ -158,14 +163,30 @@ function MoneyRequestAmountInput(
             // Remove spaces from the newAmount value because Safari on iOS adds spaces when pasting a copied value
             // More info: https://github.com/Expensify/App/issues/16974
             const newAmountWithoutSpaces = MoneyRequestUtils.stripSpacesFromAmount(newAmount);
-            const finalAmount = newAmountWithoutSpaces.includes('.')
-                ? MoneyRequestUtils.stripCommaFromAmount(newAmountWithoutSpaces)
-                : MoneyRequestUtils.replaceCommasWithPeriod(newAmountWithoutSpaces);
-            // Use a shallow copy of selection to trigger setSelection
-            // More info: https://github.com/Expensify/App/issues/16385
-            if (!MoneyRequestUtils.validateAmount(finalAmount, decimals)) {
-                setSelection((prevSelection) => ({...prevSelection}));
-                return;
+            let finalAmount: string;
+            const exactAmountWithCurrency = MoneyRequestUtils.validateAmountWithCurrency(newAmount);
+            if (exactAmountWithCurrency) {
+                const {currency: updateCurrency, amount: updateAmount} = exactAmountWithCurrency;
+                const updateDecimals = CurrencyUtils.getCurrencyDecimals(updateCurrency);
+                finalAmount = updateAmount.includes('.') ? MoneyRequestUtils.stripCommaFromAmount(updateAmount) : MoneyRequestUtils.replaceCommasWithPeriod(updateAmount);
+                // Use a shallow copy of selection to trigger setSelection
+                // More info: https://github.com/Expensify/App/issues/16385
+                if (!MoneyRequestUtils.validateAmount(finalAmount, updateDecimals)) {
+                    setSelection((prevSelection) => ({...prevSelection}));
+                    return;
+                }
+                setCurrentCurrency(updateCurrency);
+                // onPasteAmountWithCurrency?.(updateCurrency);
```

Comment:
Let's not repeat the code here. You can use this to get both the amount and the currency, and for code clarity let's always calculate `newDecimals`

```js
const [newCurrency, newAmount] = "$15.77".split(/(\d.*)/, 2)
```

make sure that newAmount is not undefined (in case of pasting just the currency)

Reply:
@s77rt In case the user only pastes currency without amount, do we update the currency?

Reply:
I don't have a strong preference, but this seems like something nice to have